src/distinguished_name.cpp

```diff
@@ -16,39 +16,47 @@
  * limitations under the License.
  * #L%
  */
-#include <tuple>
 
 #include "mococrw/distinguished_name.h"
 #include "mococrw/error.h"
 
+#include <tuple>
+
 namespace mococrw
 {
 using namespace openssl;
 
-void _addString(SSL_X509_NAME_Ptr &x509Name, const std::string &str, ASN1_NID nid)
+void _addString(SSL_X509_NAME_Ptr &x509Name, const std::string& name, const ASN1_NID nid)
```

Comment:
Wouldn't it be better to refactor this into

```c++
void _addAttribute(SSL_X509_NAME_Ptr& x509Name, const DistinguishedName::Attribute& attribute)
{
    // ...
    if (attribute.name.empty()) {
        return;
    }
    std::vector<unsigned char> vec{attribute.name.begin(), attribute.name.end()};
    _X509_NAME_add_entry_by_NID(x509Name.get(), attribute.nid, ASN1_Name_Entry_Type::ASCIIString, vec);
}
```

Of course you'd then have to support a way of returning "unspecified" from whatever you'd replace `_getAttributeByNIDAsString()` with, potentially using `std::optional` or `boost::optional`. This would make the API much cleaner, IMHO, and would also allow you to replace the series of `_addString` calls below with a loop that loops over the NIDs (plus you wouldn't have to specify the NID twice per line).

Reply:
I had it like this. But then it will break when using duplicate NIDs when using the CustomOrderBuilder. It will then always return the last entry. Now it returns the name that was actually given.

Reply:
I don't understand how this changes things for the custom order case. You're either going to do
```c++
for (const auto& it : _subjectAttributes) {
    _addString(subject, it.name, it.id);
}
```
or
```c++
for (const auto& it: _subjectAttributes) {
    _addAttribute(subject, it);
}
```
which doesn't change behavior when your `_subjectAttributes` contains duplicate NIDs. What am I missing?

Reply:
Do you like it better now?